daemon/SecretsImpl/secrets.cpp

```diff
@@ -973,13 +978,16 @@ QByteArray Daemon::ApiImpl::SecretsRequestQueue::saltData() const
     DataProtector dataProtector(saltDirPath);
     QByteArray saltData;
     DataProtector::Status s = dataProtector.getData(&saltData);
-    if (s != DataProtector::Success) {
-        qCWarning(lcSailfishSecretsDaemon) << "Can't read salt data, assuming it's corrupted. DataProtector returned:" << s;
-        qCWarning(lcSailfishSecretsDaemon) << "NOTE: If you are running a pre-release sailfish-secrets version, you need to delete all old data before proceeding";
 
-        // TODO: find an appropriate solution for dealing with data corruption.
+    if (s == DataProtector::Irretrievable) {
+        qCWarning(lcSailfishSecretsDaemon) << "saltData: salt data is irretrieveably corrupted.";
+        dealWithDataCorruption();
+    } else if (s != DataProtector::Success) {
+        qCWarning(lcSailfishSecretsDaemon) << "saltData: can't read salt data. DataProtector returned:" << s;
+        // TODO: what should we do here? The data is not corrupted, but we can't access it.
```

Comment:
Maybe we can notify the user here also, like for the data corruption case. In that case, the dealWithDataSomething() can be generalized to notify always and remove the directories only when data is corrupted.

Reply:
Maybe... I'll think about this some more.